v3.2.0 moved to `requires-python>=3.9`: "TypeError: 'type' object is not subscriptable"

### Solution to issue cannot be found in the documentation.

- [x] I checked the documentation.

### Issue

See https://github.com/microsoft/LightGBM/issues/6680.

`pyparsing` 3.2.0 dropped support for Python versions older than v3.9 (https://github.com/pyparsing/pyparsing/releases/tag/3.2.0), but packages here don't reflect that.

https://github.com/conda-forge/pyparsing-feedstock/blob/b2d4164442dd90d17be6fc75f850d6016b816de7/recipe/meta.yaml#L22

As a result, importing `pyparsing` results in the following:

```text
_________ ERROR collecting tests/python_package_test/test_utilities.py _________
tests/python_package_test/test_utilities.py:7: in <module>
    import lightgbm as lgb
../../../miniforge/envs/test-env/lib/python3.8/site-packages/lightgbm/__init__.py:11: in <module>
    from .basic import Booster, Dataset, Sequence, register_logger
../../../miniforge/envs/test-env/lib/python3.8/site-packages/lightgbm/basic.py:29: in <module>
    from .compat import (
../../../miniforge/envs/test-env/lib/python3.8/site-packages/lightgbm/compat.py:191: in <module>
    import matplotlib  # noqa: F401
../../../miniforge/envs/test-env/lib/python3.8/site-packages/matplotlib/__init__.py:129: in <module>
    from . import _api, _version, cbook, _docstring, rcsetup
../../../miniforge/envs/test-env/lib/python3.8/site-packages/matplotlib/rcsetup.py:28: in <module>
    from matplotlib._fontconfig_pattern import parse_fontconfig_pattern
../../../miniforge/envs/test-env/lib/python3.8/site-packages/matplotlib/_fontconfig_pattern.py:15: in <module>
    from pyparsing import (
../../../miniforge/envs/test-env/lib/python3.8/site-packages/pyparsing/__init__.py:129: in <module>
    from .util import *
../../../miniforge/envs/test-env/lib/python3.8/site-packages/pyparsing/util.py:14: in <module>
    class __config_flags:
../../../miniforge/envs/test-env/lib/python3.8/site-packages/pyparsing/util.py:17: in __config_flags
    _all_names: list[str] = []
E   TypeError: 'type' object is not subscriptable
```

### Installed packages

N/A

### Environment info

N/A

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -9,17 +9,17 @@ source:
   sha256: cbf74e27246d595d9a74b186b810f6fbb86726dbf3b9532efb343f6d7294fe9c
 
 build:
-  number: 0
+  number: 1
   noarch: python
   script: {{ PYTHON }} -m pip install . -vv
 
 requirements:
   host:
-    - python >=3.6
+    - python >=3.9
     - pip
-    - flit-core
+    - flit-core >=3.2,<4
   run:
-    - python >=3.6
+    - python >=3.9
 
 test:
   imports:

